[DataPipe] Fix OOM when traverse IterDataPipe due to pickling

ejguan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#72209 [DataPipe] Fix OOM when traverse IterDataPipe due to pickling**

This PR would fix https://fburl.com/jtfyksyr

Differential Revision: [D33955933](https://our.internmc.facebook.com/intern/diff/D33955933)

ejguan committed:
Fix oom

[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33955933).

NivekT reviewed (approved):
LGTM with one comment! Thanks for fixing this.

NivekT commented:
We should also add this PR to 1.11 if possible.

ejguan commented on torch/utils/data/graph.py:
```diff
@@ -28,9 +26,8 @@ def stub_pickler(obj):
-            if callable(v) or isinstance(v, PRIMITIVE):
-                continue
-            state[k] = v
+            if isinstance(v, (IterDataPipe, tuple)):
```
Makes sense

ejguan committed:
Update on "[DataPipe] Fix OOM when traverse IterDataPipe due to pickling"


This PR would fix https://fburl.com/jtfyksyr

Differential Revision: [D33955933](https://our.internmc.facebook.com/intern/diff/D33955933)

[ghstack-poisoned]

ejguan committed:
Update on "[DataPipe] Fix OOM when traverse IterDataPipe due to pickling"


This PR would fix https://fburl.com/jtfyksyr

Differential Revision: [D33955933](https://our.internmc.facebook.com/intern/diff/D33955933)

[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33955933).
